drafts/client-interface.md

```diff
@@ -1,13 +1,133 @@
 # Hydra Reference Client Interface
 
 This document describes the interface of a yet-to-be-built reference client
-library in pseudo code.
+library for ReST APIs supporting the Hydra vocabulary. In the following we call 
+that client library "Hydra client".
 
+The Hydra client works similar to a browser. You point it to a URL. The URL becomes the current resource. 
+Then you interact with the controls of the representation.
 
-## Retrieve a resource
+The examples show in pseudocode how a commandline interface application could use the Hydra client 
+for interaction with an API.
 
-Related action: [DownloadAction](http://schema.org/DownloadAction)
+## Find API entrypoint for website
+
+The Hydra vocabulary has a special mechanism to discover API entrypoints.
+
+```
+hydraClient.gotoEntrypoint(website)
+print hydraClient.representation 
+// the API entrypoint is now the current representation
+```
+
+## Retrieve a resource from URI
+
+### Using uniform interface
+
+Assuming that `apiurl` is known as the API's entrypoint and that it is a 
+[schema:FoodEstablishment](http://schema.org/FoodEstablishment) which offers a menu from which users 
+can choose food items. The menu information is not embedded in the
+`FoodEstablishment` representation, 
+rather there is only a link to the menu. I.e. the `FoodEstablishment` representation 
+has a control to retrieve the menu. 
+
+The following code assumes that the client is aware of the uniform interface of the underlying protocol. 
+In the example below the client knows that it uses the uniform interface of HTTP.
+```
+hydraClient.setLocation(apiurl);
+// apiurl is now the current representation
+
+// we want to retrieve or download the menu of the food establishment
+menu = hydraClient.controls["http://schema.org/hasMenu"]
+if(menu && menu.methods["GET"])
```

Comment:
I'm interested why the consumer should care whether it's a GET or another method? Wouldn't it be better to have it hidden behind the operation's interface. At least for the purpose of invoking/executing the operation.